CIME/case/case.py

```diff
@@ -1509,47 +1534,62 @@ def configure(
-        if max_gpus_per_node:
-            if "gpu" in compiler:
-                if not ngpus_per_node:
-                    ngpus_per_node = 1
-                    logger.warning(
-                        "Setting ngpus_per_node to 1 for compiler {}".format(compiler)
-                    )
-                expect(
-                    ngpus_per_node > 0,
-                    " ngpus_per_node is expected > 0 for compiler {}; current value is {}".format(
-                        compiler, ngpus_per_node
-                    ),
-                )
-            else:
-                expect(
-                    ngpus_per_node == 0,
-                    " ngpus_per_node is expected = 0 for compiler {}; current value is {}".format(
-                        compiler, ngpus_per_node
-                    ),
-                )
+        if gpu_type and str(gpu_type).lower() != "none":
+            expect(
+                max_gpus_per_node,
+                f"GPUS are not defined for machine={machine_name} and compiler={compiler}",
+            )
+            expect(
+                gpu_offload,
+                "Both gpu-type and gpu-offload must be defined if either is defined",
+            )
+            expect(
+                compiler in ["nvhpc", "cray"],
+                f"Only nvhpc and cray compilers are expected for a GPU run; the user given compiler is {compiler}, ",
+            )
+
+            valid_gpu_type = machobj.get_value("GPU_TYPES").split(",")
+            if "none" in valid_gpu_type:
+                valid_gpu_type.remove("none")
+            expect(
+                gpu_type in valid_gpu_type,
+                f"Unsupported GPU type is given: {gpu_type} ; valid values are {valid_gpu_type}",
+            )
+            self.gpu_enabled = True
             if ngpus_per_node >= 0:
                 self.set_value(
                     "NGPUS_PER_NODE",
-                    ngpus_per_node
+                    max(1, ngpus_per_node)
                     if ngpus_per_node <= max_gpus_per_node
                     else max_gpus_per_node,
                 )
+            valid_gpu_offload = machobj.get_value("GPU_OFFLOAD").split(",")
+            if "none" in valid_gpu_offload:
+                valid_gpu_offload.remove("none")
+            expect(
+                gpu_offload in valid_gpu_offload,
+                f"Unsupported GPU offload is given: {gpu_offload} ; valid values are {valid_gpu_offload}",
+            )
+
+        elif gpu_offload and str(gpu_offload).lower() != "none":
+            expect(
+                False,
+                "Both gpu-type and gpu-offload must be defined if either is defined",
+            )
+
+        # Set these two GPU XML variables here to overwrite the default values
+        if gpu_type:
+            self.set_value("GPU_TYPE", gpu_type)
+
+        if gpu_offload and gpu_offload != "none":
+            self.set_value("GPU_OFFLOAD", gpu_offload)
```

Comment:
Interesting. In my version, I do not have these `if` statements (see https://github.com/sjsprecious/cime/blob/e3607d1ff76e78f2d41203e88267ba8d36b6bf1f/CIME/case/case.py#L1584) and I always set `GPU_TYPE` and `GPU_OFFLOAD` based on the input values. Will my version be more straightforward?

Reply:
Have you tested cases in which you are not using gpu's at all?  I ran into this when compiling on gust using nvhpc but for a cpu case.

Reply:
Hi @jedwards4b , thanks for your quick reply. My version works for NVHPC on CPU on Gust (i.e., send the job to a 128-CPU node rather than 64-CPU node). My command looks `./create_newcase --case XXXXX --mach gust --compiler nvhpc --mpilib mpich --compset F2000dev --res f19_f19_mg17 --walltime 01:00:00 --queue main --project XXXX`. Could you share what error you encountered in your case?